File: programmers/js/level00/120852.js
// 소인수분해
function solution(n) {
  const answer = [];
  const isPrime = (num) => {       
    for (let i = 2; i <= Math.sqrt(num); i++) {
      if (num % i === 0) {
        return false;
      }
    }
    
    return true;
  }
  
  for (let i = 2; i <= n; i++) {
    if (n % i === 0 && isPrime(i)) {
      answer.push(i);
    }
  }
  
  return answer;
}

File: programmers/java/level00/120852.java
// 소인수분해

import java.util.stream.IntStream;

class Solution {
    boolean isPrime(int num) {
        if (num < 2) return false;
        
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) return false;
        }
        
        return true;
    }
    
    public int[] solution(int n) {
        return IntStream.range(1, n + 1)
            .filter(num -> n % num == 0 && isPrime(num))
            .toArray();
    }
}